core/src/banking_stage/transaction_scheduler/greedy_scheduler.rs

```diff
@@ -0,0 +1,720 @@
+                &mut self.account_locks,
+                schedulable_threads,
+                |thread_set| {
+                    PrioGraphScheduler::<Tx>::select_thread(
+                        thread_set,
+                        &batches.total_cus,
+                        self.in_flight_tracker.cus_in_flight_per_thread(),
+                        &batches.transactions,
+                        self.in_flight_tracker.num_in_flight_per_thread(),
+                    )
+                },
+            ) {
+                Err(TransactionSchedulingError::Filtered) => {
+                    container.remove_by_id(id.id);
+                }
+                Err(TransactionSchedulingError::UnschedulableConflicts) => {
+                    num_unschedulable += 1;
+                    self.unschedulables.push(id);
+                }
+                Ok(TransactionSchedulingInfo {
+                    thread_id,
+                    transaction,
+                    max_age,
+                    cost,
+                }) => {
+                    assert!(
+                        self.working_account_set.take_locks(&transaction),
+                        "locks must be available"
+                    );
+                    saturating_add_assign!(num_scheduled, 1);
+                    batches.transactions[thread_id].push(transaction);
+                    batches.ids[thread_id].push(id.id);
+                    batches.max_ages[thread_id].push(max_age);
+                    saturating_add_assign!(batches.total_cus[thread_id], cost);
+
+                    // If target batch size is reached, send all the batches
+                    if batches.ids[thread_id].len() >= self.config.target_transactions_per_batch {
+                        saturating_add_assign!(num_sent, self.send_batches(&mut batches,)?);
+                        self.working_account_set.clear();
+                    }
+
+                    // if the thread is at max_cu_per_thread, remove it from the schedulable threads
+                    // if there are no more schedulable threads, stop scheduling.
+                    if self.in_flight_tracker.cus_in_flight_per_thread()[thread_id]
+                        + batches.total_cus[thread_id]
+                        >= max_cu_per_thread
+                    {
+                        schedulable_threads.remove(thread_id);
+                        if schedulable_threads.is_empty() {
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+
+        self.working_account_set.clear();
+        self.send_batches(&mut batches)?;
+
+        // Push unschedulables back into the queue
+        for id in self.unschedulables.drain(..) {
+            container.push_id_into_queue(id);
+        }
+
+        Ok(SchedulingSummary {
+            num_scheduled,
+            num_unschedulable,
+            num_filtered_out,
+            filter_time_us: total_filter_time_us,
```

Comment:
`total_filter_time_us` isn't assigned?

Reply:
yeah should just set this directly to 0, there's no filter going on because that's a pre_graph filter, and there's no graph.

Reply:
6d71f6a182e806d6daa22dbb0805360b5299fd2a